core/src/main/java/org/apache/calcite/avatica/AvaticaConnection.java

```diff
@@ -392,14 +420,17 @@ public void setClientInfo(Properties properties)
   }
 
   public String getClientInfo(String name) throws SQLException {
-    throw HELPER.unsupported();
+    checkOpen();
+    return null;
```

Comment:
This could be `return this.getClientInfo().getProperty(name)` now that `getClientInfo()` returns a `Properties` object. This would remove the need for the added `checkOpen()` and explicit `return null`.